src/components/dialogs/network-modification/composite-modification/composite-modification-edition-dialog.tsx

```diff
@@ -0,0 +1,169 @@
+ */
+import { FunctionComponent, SyntheticEvent, useEffect, useState } from 'react';
+import { useParameterState } from '../../use-parameters-dialog';
+import { PARAM_LANGUAGE } from '../../../../utils/config-params';
+import {
+    CustomMuiDialog,
+    FieldConstants,
+    NetworkModificationMetadata,
+    NO_SELECTION_FOR_COPY,
+    unscrollableDialogStyles,
+    useModificationLabelComputer,
+    useSnackMessage,
+    yupConfig as yup,
+} from '@gridsuite/commons-ui';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { getCompositeModificationContent, saveCompositeModification } from '../../../../utils/rest-api';
+import CompositeModificationEditionForm from './composite-modification-edition-form';
+import { List, ListItem } from '@mui/material';
+import { useIntl } from 'react-intl';
+import Divider from '@mui/material/Divider';
+import Box from '@mui/material/Box';
+import { setSelectionForCopy } from '../../../../redux/actions';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppState } from '../../../../redux/reducer';
+
+const schema = yup.object().shape({
+    [FieldConstants.NAME]: yup.string().trim().required('nameEmpty'),
+});
+
+const emptyFormData = (name?: string) => ({
+    [FieldConstants.NAME]: name,
+});
+
+interface FormData {
+    [FieldConstants.NAME]: string;
+}
+
+interface CompositeModificationEditionDialogProps {
+    compositeModificationId: string;
+    open: boolean;
+    onClose: (event?: SyntheticEvent) => void;
+    titleId: string;
+    name: string;
+    broadcastChannel: BroadcastChannel;
+}
+
+export const CompositeModificationEditionDialog: FunctionComponent<CompositeModificationEditionDialogProps> = ({
+    compositeModificationId,
+    open,
+    onClose,
+    titleId,
+    name,
+    broadcastChannel,
+}: Readonly<CompositeModificationEditionDialogProps>) => {
+    const intl = useIntl();
+    const [languageLocal] = useParameterState(PARAM_LANGUAGE);
+    const [isFetching, setIsFetching] = useState(!!compositeModificationId);
+    const { snackError } = useSnackMessage();
+    const selectionForCopy = useSelector((state: AppState) => state.selectionForCopy);
+    const [modifications, setModifications] = useState<NetworkModificationMetadata[]>([]);
+    const dispatch = useDispatch();
+
+    const methods = useForm<FormData>({
+        defaultValues: emptyFormData(name),
+        resolver: yupResolver(schema),
+    });
+
+    const { computeLabel } = useModificationLabelComputer();
+    const getModificationLabel = (modif: NetworkModificationMetadata) => {
+        if (!modif) {
+            return null;
+        }
+        return intl.formatMessage(
+            { id: 'network_modifications.' + modif.type },
+            {
+                ...modif,
+                ...computeLabel(modif),
+            }
+        );
+    };
+
+    const renderNetworkModificationsList = () => {
```

Comment:
render.. could be improved, it doesn't clearly convey what the function does 
maybe (generate or create)NetworkModificationsList or something else

Reply:
This is just a copy paste. Maybe the original author used "render" because it contains jsx anchor code ?
Anyway I prefer generate so ok.